fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/AdminCompactTableCommand.java

```diff
@@ -0,0 +1,126 @@
+    private TableRefInfo tableRefInfo;
+    private EqualTo where;
+
+    /**
+     * compact type
+     */
+    public enum CompactionType {
+        CUMULATIVE,
+        BASE
+    }
+
+    private CompactionType typeFilter;
+
+    public AdminCompactTableCommand(TableRefInfo tableRefInfo, EqualTo where) {
+        super(PlanType.ADD_CONSTRAINT_COMMAND);
+        this.tableRefInfo = tableRefInfo;
+        this.where = where;
+    }
+
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        validate(ctx);
+        String dbName = tableRefInfo.getTableNameInfo().getDb();
+        String tableName = tableRefInfo.getTableNameInfo().getTbl();
+        String type = getCompactionType();
+        List<String> partitionNames = tableRefInfo.getPartitionNamesInfo().getPartitionNames();
+        ctx.getEnv().compactTable(dbName, tableName, type, partitionNames);
+    }
+
+    private void validate(ConnectContext ctx) throws UserException {
+        // check auth
+        if (!Env.getCurrentEnv().getAccessManager().checkGlobalPriv(ConnectContext.get(), PrivPredicate.ADMIN)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_SPECIFIC_ACCESS_DENIED_ERROR, "ADMIN");
+        }
+        tableRefInfo.analyze(ctx);
+        Util.prohibitExternalCatalog(tableRefInfo.getTableNameInfo().getCtl(), this.getClass().getSimpleName());
+
+        List<String> partitionNames = tableRefInfo.getPartitionNamesInfo().getPartitionNames();
+        if (partitionNames != null) {
+            if (partitionNames.size() != 1) {
+                throw new AnalysisException("Only support single partition for compaction");
+            }
+        } else {
+            throw new AnalysisException("No partition selected for compaction");
+        }
+
+        // analyze where clause if not null
+        if (where == null) {
+            throw new AnalysisException("Compaction type must be specified in"
+                + " Where clause like: type = 'BASE/CUMULATIVE'");
+        }
+
+        if (!analyzeWhere()) {
+            throw new AnalysisException(
+                "Where clause should looks like: type = 'BASE/CUMULATIVE'");
+        }
+    }
+
+    private boolean analyzeWhere() {
+        try {
+            typeFilter = CompactionType.valueOf(((StringLiteral) where.right()).getStringValue().toUpperCase());
+        } catch (Exception e) {
+            return false;
+        }
+
+        if (typeFilter == null || (typeFilter != CompactionType.CUMULATIVE && typeFilter != CompactionType.BASE)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context) {
+        return visitor.visitAdminCompactTableCommand(this, context);
+    }
+
+    private String getCompactionType() {
+        if (typeFilter == CompactionType.CUMULATIVE) {
+            return "cumulative";
+        } else {
+            return "base";
+        }
+    }
+}
```

Comment:
override checkSupportedInCloudMode method and implement as checkDdlStmtSupported method in DdlExecutor.java

Reply:
done